cpp/src/parquet/stream_reader.cc

```diff
@@ -36,6 +36,20 @@ StreamReader::StreamReader(std::unique_ptr<ParquetFileReader> reader)
   NextRowGroup();
 }
 
+int StreamReader::num_columns() const {
+  if (file_metadata_) {
+    return file_metadata_->num_columns();
+  }
+  return 0;
+}
+
+int64_t StreamReader::num_rows() const {
+  if (file_metadata_) {
+    return file_metadata_->num_rows();
+  }
+  return 0;
```

Comment:
Is this only for the null initialization case?  Would it be better to throw an exception?

Reply:
Yes the test is only for null initialization.
It is possible to return the true value for the number of rows i.e. zero so there is no point in throwing an exception.